storage/archive/ha_archive.cc

```diff
@@ -990,7 +990,7 @@ int ha_archive::write_row(const uchar *buf)
 
   if (!share->archive_write_open && share->init_archive_writer())
   {
-    rc= errno;
+    rc= HA_ERR_CRASHED_ON_USAGE;
```

Comment:
Please don't change errno to HA_ERR_CRASHED_ON_USAGE.
This will hide the original error !